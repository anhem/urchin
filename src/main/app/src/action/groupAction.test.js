import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import {
    addUser,
    createGroup,
    deleteGroup,
    getGroup,
    getGroups,
    getUsersForGroup,
    removeUser,
    setGroup
} from "./groupAction";

describe('groupAction', () => {

    const getGroupsResponse = {};
    let groupId = 10;
    const userId = 100;
    const group = {groupName: 'name'};
    let store;

    beforeEach(() => {
        store = configureMockStore([thunk])({});
    });

    afterEach(() => {
        fetchMock.reset();
        fetchMock.restore();
    });

    it('getGroups is successful', () => {

        fetchMock.getOnce('/api/groups', getGroupsResponse);

        return store.dispatch(getGroups()).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('getGroup is successful', () => {

        fetchMock.getOnce('/api/groups/' + groupId, getGroupsResponse);

        return store.dispatch(getGroup(groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('setGroup is successful', () => {

        store.dispatch(setGroup(group));
        expect(store.getActions()).toMatchSnapshot();
    });

    it('createGroup is successful', () => {

        fetchMock.postOnce('/api/groups/add', group);

        return store.dispatch(createGroup(group)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('createGroup failed', () => {

        return store.dispatch(createGroup(group)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('deleteGroup is successful', () => {

        fetchMock.deleteOnce('/api/groups/' + groupId, {});

        return store.dispatch(deleteGroup(groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('deleteGroup failed', () => {

        return store.dispatch(deleteGroup(groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('getUsersForGroup is successful', () => {

        fetchMock.getOnce('/api/groups/' + groupId + '/users', getGroupsResponse);

        return store.dispatch(getUsersForGroup(groupId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('addUser is successful', () => {

        let data = {
            userId: userId,
            groupId: groupId
        };

        fetchMock.postOnce('/api/groups/user', data);
        fetchMock.getOnce('/api/groups/' + groupId + '/users', getGroupsResponse);

        return store.dispatch(addUser(groupId, userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('addUser failed', () => {

        return store.dispatch(addUser(groupId, userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('removeUser is successful', () => {

        fetchMock.deleteOnce('/api/groups/' + groupId + "/user/" + userId, {});
        fetchMock.getOnce('/api/groups/' + groupId + '/users', getGroupsResponse);

        return store.dispatch(removeUser(groupId, userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });

    it('removeUser failed', () => {

        return store.dispatch(removeUser(groupId, userId)).then(() => {
            expect(store.getActions()).toMatchSnapshot();
        })
    });
});