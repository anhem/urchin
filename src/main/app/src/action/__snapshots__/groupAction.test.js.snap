// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`groupAction addUser failed 1`] = `
Array [
  Object {
    "type": "ADD_USER_TO_GROUP",
  },
]
`;

exports[`groupAction addUser is successful 1`] = `
Array [
  Object {
    "type": "ADD_USER_TO_GROUP",
  },
  Object {
    "data": Object {
      "groupId": 10,
      "userId": 100,
    },
    "type": "ADD_USER_TO_GROUP_SUCCESS",
  },
  Object {
    "type": "GET_USERS_FOR_GROUP",
  },
  Object {
    "data": Object {},
    "type": "GET_USERS_FOR_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction createGroup failed 1`] = `
Array [
  Object {
    "type": "SAVE_GROUP",
  },
]
`;

exports[`groupAction createGroup is successful 1`] = `
Array [
  Object {
    "type": "SAVE_GROUP",
  },
  Object {
    "data": Object {
      "groupName": "name",
    },
    "type": "SAVE_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction deleteGroup failed 1`] = `
Array [
  Object {
    "type": "DELETE_GROUP",
  },
]
`;

exports[`groupAction deleteGroup is successful 1`] = `
Array [
  Object {
    "type": "DELETE_GROUP",
  },
  Object {
    "data": Object {},
    "type": "DELETE_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction getGroup is successful 1`] = `
Array [
  Object {
    "type": "GET_GROUP",
  },
  Object {
    "data": Object {},
    "type": "GET_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction getGroups is successful 1`] = `
Array [
  Object {
    "type": "GET_GROUPS",
  },
  Object {
    "data": Object {},
    "type": "GET_GROUPS_SUCCESS",
  },
]
`;

exports[`groupAction getUsersForGroup is successful 1`] = `
Array [
  Object {
    "type": "GET_USERS_FOR_GROUP",
  },
  Object {
    "data": Object {},
    "type": "GET_USERS_FOR_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction removeUser failed 1`] = `
Array [
  Object {
    "type": "REMOVE_USER",
  },
]
`;

exports[`groupAction removeUser is successful 1`] = `
Array [
  Object {
    "type": "REMOVE_USER",
  },
  Object {
    "data": Object {},
    "type": "REMOVE_USER_SUCCESS",
  },
  Object {
    "type": "GET_USERS_FOR_GROUP",
  },
  Object {
    "data": Object {},
    "type": "GET_USERS_FOR_GROUP_SUCCESS",
  },
]
`;

exports[`groupAction setGroup is successful 1`] = `
Array [
  Object {
    "data": Object {
      "groupName": "name",
    },
    "type": "SET_GROUP",
  },
]
`;
